oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/tar/TarWriter.java

```diff
@@ -183,7 +183,14 @@ void flush() throws IOException {
             }
 
             if (doFlush) {
-                archive.flush();
+                archive.flush(() -> {
+                    try {
+                        writeBinaryReferences();
+                        writeGraph();
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
```

Comment:
Probably more specific exception should be thrown.

Reply:
An alternative would be to declare a functional interface declaring an `IOException`:

```java
interface OnFlushSuccessCallback {
  void run() throws IOException;
}
```
Then use it instead of `Runnable`:
```java
default void flush(OnFlushSuccessCallback onSuccess) throws IOException
```
With this, no need to catch the `IOException` to wrap it in a `RuntimeException` :
```java
 archive.flush(() -> {
     writeBinaryReferences();
     writeGraph();
});
```